### DISCUSSION TITLE
Scaling Z axis within the code

### DISCUSSION THREAD

**Discussion by rajat2g** [pyvista/pyvista#2021] [Q&A]
Hello,
I am looking to scale the Z axis for my visualizations. I have a points to tube structure within the visualization. Scaling in Z makes it look like a vertical ellipse. Without changing the scale resolution in the X and Y direction, how can I automate the change in tube geometry with the change in Z scale (from the menu bar)? Attached are the two images with examples. 

Appreciate the help.

![image](https://user-images.githubusercontent.com/97545251/148990445-3d00e51a-7fec-47f6-86aa-f3dfae6e3030.png)
![image](https://user-images.githubusercontent.com/97545251/148990504-4a080ae2-c589-4061-bd08-69cd6dcdf262.png)

**Comment by tkoyama010** [comment]
PyVista is just displaying the modified mesh. Therefore, if you want to change the resolution of the mesh, you will have to change the mesh itself. If the mesh was created by code, the remeshing could be automated. Can you share the code for mesh creation?

**Reply by rajat2g** [reply]
Thanks for the response. Below is a screenshot of the file I read and the mesh code to make it a cylinder.

![image](https://user-images.githubusercontent.com/97545251/149607678-2757841b-3be0-40e4-9475-06d26dd37da8.png)

```py
spat_ref = ['Easting', 'Northing', 'Tun-Elevation']
tunnel_ring_pts = pv.PolyData(tunnel_var[spat_ref].values)
tunnel_ch_pts = pv.PolyData(tunnel_var[spat_ref].values)
tunnel_ring_pts['Ring'] = tunnel_var['Ring'].values
tunnel_ch_pts['Chainage'] = tunnel_var['Chainage'].values
# Make tubes/tunnels
r = 19
tunnelfy = lambda pts: PVGeo.filters.PointsToTube(radius=r).apply(pts)
tunnel_ci_ring = tunnelfy(tunnel_ring_pts)
tunnel_ci_ch = tunnelfy(tunnel_ch_pts)
```

**Reply by tkoyama010** [reply]
Hmm. It seems to use PVGeo. I am not familiar with it. @banesullivan Any ideas?

**Reply by banesullivan** [reply]
> Scaling in Z makes it look like a vertical ellipse.

This is expected - leaving X and Y constant, scaling a tube in the Z direction should do this

> Without changing the scale resolution in the X and Y direction, how can I automate the change in tube geometry with the change in Z scale (from the menu bar)?

What kind of change in geometry are you hoping to automate?
